packages/vega-parser/src/util.js

```diff
@@ -61,7 +61,12 @@ export function sortKey(sort) {
 export function aggrField(op, field) {
   return (op && op.signal ? '$' + op.signal : op || '')
     + (op && field ? '_' : '')
-    + (field && field.signal ? '$' + field.signal : field || '');
+    + (field && field.signal
+        ? '$' + field.signal
+        // Replace non-alphanumeric character sequences with underscores and trim leading/trailing underscores
```

Comment:
What's wrong with trailing or leading _, btw?

Reply:
Without trimming the name of the aggregated field would differ depending on the notation of the target field: `'c\\.d'` =would become `'sum_c_d'` and  `'[c.d]'` would become  `'sum___c_d_'`, which I think is undesirable.